src/hooks/VoicePlayer/dux/reducer.ts

```diff
@@ -0,0 +1,108 @@
+import {
+  INITIALIZE_AUDIO_UNIT,
+  ON_CURRENT_TIME_UPDATE,
+  ON_VOICE_PLAYER_PAUSE,
+  ON_VOICE_PLAYER_PLAY,
+  SET_CURRENT_PLAYER,
+} from "./actionTypes";
+import {
+  AudioStorageUnit,
+  AudioUnitDefaultValue,
+  VoicePlayerInitialState,
+  VoicePlayerStatus,
+} from "./initialState";
+
+type InitializeAudioUnitPayload = { groupKey: string };
+type SetCurrentPlayerPayload = { audioPlayer: HTMLAudioElement, groupKey: string };
+type OnVoicePlayerPlayPayload = { groupKey: string, audioFile: File };
+type OnVoicePlayerPausePayload = { groupKey: string };
+type OnCurrentTimeUpdatePayload = { groupKey: string };
+type PayloadType = (
+  InitializeAudioUnitPayload
+  | SetCurrentPlayerPayload
+  | OnVoicePlayerPlayPayload
+  | OnVoicePlayerPausePayload
+  | OnCurrentTimeUpdatePayload
+);
```

Comment:
I'm not sure if this is a good idea, here too

Reply:
This can be made cleaner ~ keep this for now.. I will suggest after you release